api/lib/opentelemetry/correlation_context/propagation/http_extractor.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'cgi'
+
+module OpenTelemetry
+  module CorrelationContext
+    module Propagation
+      # Extracts correlations from carriers in the W3C Correlation Context format
+      class HttpExtractor
+        include Context::Propagation::DefaultGetter
+
+        # Returns a new HttpExtractor that extracts context using the specified
+        # header key
+        #
+        # @param [String] correlation_context_key The correlation context header
+        #   key used in the carrier
+        # @return [HttpExtractor]
+        def initialize(correlation_context_key: 'Correlation-Context')
+          @correlation_context_key = correlation_context_key
+        end
+
+        # Extract remote correlations from the supplied carrier.
+        # If extraction fails, the original context will be returned
+        #
+        # @param [Context] context The context to be updated with extracted correlations
+        # @param [Carrier] carrier The carrier to get the header from
+        # @param [optional Callable] getter An optional callable that takes a carrier and a key and
+        #   returns the value associated with the key. If omitted the default getter will be used
+        #   which expects the carrier to respond to [] and []=.
+        # @yield [Carrier, String] if an optional getter is provided, extract will yield the carrier
+        #   and the header key to the getter.
+        # @return [Context] context updated with extracted correlations, or the original context
+        #   if extraction fails
```

Comment:
It is not clear to me from reading the code that we ever return `the original context if extraction fails`.

Reply:
I added code to make the comment accurate 🙂.